### ISSUE TITLE
Issues in InvoiceResource

### ISSUE THREAD

**Issue by reshmabidikar** [killbill/killbill#1582]
- An extra `locale` parameter is specified in the [InvoiceResource.getInvoiceMPTemplate](https://github.com/killbill/killbill/blob/585d68891b06f3329379548583a33fe94b9250ea/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceResource.java#L801) method definition. This needs to be removed. Once this is fixed, the code for the [Kill Bill Java client](https://github.com/killbill/killbill-client-java) also needs to be regenerated.
- The Java code corresponding to the [Retrieve invoice item audit logs with history](https://killbill.github.io/slate/#invoice-retrieve-invoice-item-audit-logs-with-history-by-invoice-item-id) and [Retrieve invoice payment audit logs with history](https://killbill.github.io/slate/#invoice-retrieve-invoice-payment-audit-logs-with-history-by-invoice-payment-id) is missing from [InvoiceResource](https://github.com/killbill/killbill/blob/585d68891b06f3329379548583a33fe94b9250ea/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceResource.java). This needs to be added and the code for the [Kill Bill Java client](https://github.com/killbill/killbill-client-java) needs to be regenerated.
- There is a typo in the error message specified in the `InvoiceResource. generateDryRunInvoice` [here](https://github.com/killbill/killbill/blob/585d68891b06f3329379548583a33fe94b9250ea/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceResource.java#L401). So, when the `product name` is missing, the error message indicates that the `product category` is not specified.
- The [Upload Manual Pay Template](https://killbill.github.io/slate/#invoice-upload-the-manualpay-invoice-template-for-the-tenant) endpoint returns a location header containing a `locale` while the [Upload Invoice Template](https://killbill.github.io/slate/#invoice-upload-the-invoice-template-for-the-tenant) does not return a `locale` in the location header. In general, locale does not make sense for templates since templates are independent of locales, so this needs to be investigated further.

**Comment by sbrossie** [comment]
> The Java code corresponding to the [Retrieve invoice item audit logs with history](https://killbill.github.io/slate/#invoice-retrieve-invoice-item-audit-logs-with-history-by-invoice-item-id) and [Retrieve invoice payment audit logs with history](https://killbill.github.io/slate/#invoice-retrieve-invoice-payment-audit-logs-with-history-by-invoice-payment-id) is missing from [InvoiceResource](https://github.com/killbill/killbill/blob/585d68891b06f3329379548583a33fe94b9250ea/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceResource.java). This needs to be added and the code for the [Kill Bill Java client](https://github.com/killbill/killbill-client-java) needs to be regenerated.

Aren't those exposed in our [InvoiceItemResource](https://github.com/killbill/killbill/blob/585d68891b06f3329379548583a33fe94b9250ea/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceItemResource.java#L97) ?



\cc @reshmabidikar

**Comment by reshmabidikar** [comment]
> Aren't those exposed in our [InvoiceItemResource](https://github.com/killbill/killbill/blob/585d68891b06f3329379548583a33fe94b9250ea/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoiceItemResource.java#L97) ?

Yes, looks like it. Actually, the ticket is old and the links in the ticket no longer work (probably since we restructured our slate docs). So let's ignore this one.
